src/crawlee/_autoscaling/snapshotter.py

```diff
@@ -94,16 +104,20 @@ def __init__(
             cast(float, available_memory_ratio)
         )
 
-        self._cpu_snapshots: list[CpuSnapshot] = []
-        self._event_loop_snapshots: list[EventLoopSnapshot] = []
-        self._memory_snapshots: list[MemorySnapshot] = []
-        self._client_snapshots: list[ClientSnapshot] = []
+        self._cpu_snapshots: SortedList[CpuSnapshot] = self._get_sorted_list_by_created_at([])
+        self._event_loop_snapshots: SortedList[EventLoopSnapshot] = self._get_sorted_list_by_created_at([])
+        self._memory_snapshots: SortedList[MemorySnapshot] = self._get_sorted_list_by_created_at([])
+        self._client_snapshots: SortedList[ClientSnapshot] = self._get_sorted_list_by_created_at([])
```

Comment:
Maybe we could init it in this way and benefit from type inference:

```python
        # ...
        self._cpu_snapshots = self._create_sorted_snapshot_list(CpuSnapshot)
        self._event_loop_snapshots = self._create_sorted_snapshot_list(EventLoopSnapshot)
        self._memory_snapshots = self._create_sorted_snapshot_list(MemorySnapshot)
        self._client_snapshots = self._create_sorted_snapshot_list(ClientSnapshot)
        # ...

   @staticmethod
    def _create_sorted_snapshot_list(_snapshot_type: T) -> SortedList[T]:
        return SortedList(list[T](), key=attrgetter('created_at'))
```

Reply:
I did it halfway :-)  as I am using this function a lot in tests. 
Other option would be to define some helper function in tests and keep your suggestion. Do you have strong preference?

Reply:
Nooo, this is good, thanks :slightly_smiling_face:.